src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -0,0 +1,846 @@
+            /// For example, if user typed "C", roslyn might select "CancellationToken", but with suggested default "Console" we will end up selecting "Console" instead.
+            /// </summary>
+            private ItemSelection GetDefaultsMatch(IReadOnlyList<MatchResult<VSCompletionItem>> items, ItemSelection intialSelection)
+            {
+                // Because the items are already sorted based on pattern-matching score, try to limit the range for the items we compare default with
+                // by searching for the first "inferior" item, so we can avoid always going through the entire list.
+                int inferiorItemIndex;
+                if (_filterText.Length == 0)
+                {
+                    // Without filterText, all items are equally good match (w.r.t to the empty filterText), so we have to consider all of them.
+                    inferiorItemIndex = items.Count;
+                }
+                else
+                {
+                    // Because the items are sorted based on pattern-matching score, the selectedIndex is in the middle of a range of
+                    // -- as far as the pattern matcher is concerned -- equivalent items (items with identical PatternMatch.Kind and IsCaseSensitive).
+                    // Find the last items in the range and use that to limit the items searched for from the defaults list.          
+                    var selectedItemMatch = items[intialSelection.SelectedItemIndex].PatternMatch;
+
+                    if (!selectedItemMatch.HasValue)
+                        return intialSelection;
+
+                    inferiorItemIndex = intialSelection.SelectedItemIndex;
+                    while (++inferiorItemIndex < items.Count)
+                    {
+                        var itemMatch = items[inferiorItemIndex].PatternMatch;
+                        if (!itemMatch.HasValue
+                            || itemMatch.Value.Kind != selectedItemMatch.Value.Kind
+                            || itemMatch.Value.IsCaseSensitive != selectedItemMatch.Value.IsCaseSensitive)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                foreach (var defaultText in _data.Defaults)
+                {
+                    // The range includes all items that are as good of a match as what we intially selected (and in descending order of matching score)
+                    // so we just need to search for the first item that matches the suggested default.
+                    for (var i = 0; i < inferiorItemIndex; ++i)
+                    {
+                        if (items[i].RoslynCompletionItem.DisplayText == defaultText)
+                            return intialSelection with { SelectedItemIndex = i };
+                    }
+                }
+
+                return intialSelection;
+            }
```

Comment:
tbh, i didn't really understand this method.  actual concrete examples would be useful.